### PR TITLE
Fix enclitics preprocessing

### PR DESCRIPTION
**Pull request by arthurian**
This PR fixes a small issue with the enclitics preprocessing step and does some general cleanup.

Changes:
- Updated preprocessor to ensure that when an enclitic is broken off and two triples are returned, one for the base word and one for the enclitic, the `word_normalized` in the first triple includes the base word and not the enclitic. Otherwise, you end up with a situation where token `estne` is tokenized as `('est', 'ne', ''),('ne', 'ne', ' ')` when it should be `('est', 'est', ''),('ne', 'ne', ' ')`.
- Added tokenizer and preprocessor unit tests.
- General cleanup (e.g. removed `LatinLexicon` model).

### CODE REVIEW

**Review by elliottyates — COMMENTED** [review]
Just one comment on some edge cases @arthurian -- also, have you noticed the cool beta feature "unchanged files with check annotations" at the bottom of the `/files` page? That's neat! I've never noticed that before (nor does it appear in the "navigation" on the left -- it seems to be almost like an easter egg)...

File: lemmatization/services/latin.py [hunk 1]
```diff
@@ -159,8 +143,8 @@ def preprocessor(self, tokens):
                     for enclitic in LATIN_ENCLITICS:
                         if word.endswith(enclitic) and found_first_enclitic is False:
                             found_first_enclitic = True
-                            #Creates example word, word_normalized, following = new_token
-                            new_tokens.append((word[:word.find(enclitic)], enclitic, ""))
+                            word_without_enclitic = word[:word.find(enclitic)]
```
[the remark this answers is not in the document]
Comment (Tanvez) [reply lemmatization/services/latin.py:146]: Agreed with your points @elliottyates There could be the edge case that you pointed out where the `enclitic pattern` would appear other places(names?) but I am not totally sure. @bilbe might know more about this. I think for now we are expecting the enclitics to only appear at the end of the word which this piece of code(`word[:-len(enclitic)]`) should do to slice off the enclitic.

**Review by Tanvez — APPROVED** [review]
Looks good @arthurian ! Left a comment piggy backing off of @elliottyates comments about the enclitic slice function edge case

**Comment by elliottyates** [comment]
Just curious @arthurian or @bilbe was a more specific tokenizer ever investigated, like e.g. [this one](https://docs.cltk.org/en/latest/cltk.tokenizers.html#cltk.tokenizers.processes.LatinTokenizationProcess) from the [CLTK](https://docs.cltk.org/en/latest/about.html) ([github](https://github.com/cltk/cltk))?

File: lemmatization/services/latin.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaguillette) [reply lemmatization/services/latin.py:146]: The `word[:-len(enclitic)]` approach makes sense to me, too. Maybe to handle proper nouns `word_without_enclitic` should go through `lookup_form()` as well, and if there's no found form with the enclitic removed, it just gives up and returns the token as is? I feel like any time I'm not sure if there's an edge case in a language, there is, especially with user-generated input.

**Review by jaguillette — APPROVED** [review]
Looks good to me overall, but I think that the edge cases Elliott brought up could still be a concern. I added comments there, but everything else looks great!

File: lemmatization/services/latin.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (arthurian) [reply lemmatization/services/latin.py:146]: Great catch @elliottyates @Tanvez @jaguillette . Thank you for spotting this! Yes, this is definitely an edge case that we need to handle. Instead of searching from the start of the string for the first occurrence of the enclitic, we could start from the end and work backward as suggested since the enclitic is a known string. I agree as well that we might erroneously split a token that appears to have the enclitic, even though it's not, such as in a person's name. Good idea @jaguillette to only split if we can positively identify the form, otherwise leave it unchanged. I'm going to rework this with all of your suggestions.

**Comment by arthurian** [comment]
@elliottyates I recall we had looked at CLTK a few years ago and decided against using it, for reasons that are not documented anywhere and are perhaps worth revisiting. @bilbe might be able to shed some more light on that, but either way, I think it's worth another look now that we're getting into more than basic tokenization.

**Comment by elliottyates** [comment]
@arthurian makes sense -- definitely an experience we have been through before -- please let me know if I can help in evaluating it in any way! It (and possibly other libraries in the field) looks to be continually maintained, which is a nice plus.
